moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -557,7 +557,7 @@ class PlanningSceneMonitor : private boost::noncopyable
   // Lock for state_update_pending_ and dt_state_update_
```

Comment:
```suggestion
  // Lock for last_robot_state_update_wall_time_ and dt_state_update_
```
from what I see here, the mutex does not actually need to protect `state_update_pending_`, although it does in many places. Please proof me wrong.

Either way at least it guarding the `update_wall_time` should be mentioned here.

Reply:
I agree adding `dt_state_update_` to the list. However, state_update_pending_ should be kept too.
I guess you refer here to the following snippet of your main comment:

> onStateUpdate, all direct ROS callbacks the CSM and PSM use, and stateUpdateTimerCallback are all served from a single threaded AsyncSpinner. Thus they are effectively serialized and don't need a mutex at all.

However, you also state:

> `waitForCurrentState()` can call (the public API) `updateSceneWithCurrentState` from outside the spinner and can thus affect `last_robot_state_update_wall_time_`, requiring the mutex.

But `updateSceneWithCurrentState` modifies both, `state_update_pending_` and `last_robot_state_update_wall_time_`.

Reply:
Yes, but there is no real consistency required between the two, so consistency between multiple variables is not an argument.
`state_update_pending_` is volatile and (almost everywhere) single-instruction.

For proper correctness, aka no formal data races, you need to add the mutex also [here](https://github.com/moveit/moveit/blob/9183ad9f5a208562ddcd562d1a8221dba6e368da/moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp#L933) and [here](https://github.com/moveit/moveit/blob/9183ad9f5a208562ddcd562d1a8221dba6e368da/moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp#L1195) to the checks.